syft/formats/common/spdxhelpers/to_syft_model.go

```diff
@@ -353,6 +353,7 @@ func extractMetadata(p *spdx.Package2_2, info pkgInfo) (pkg.MetadataType, interf
 		}
 		return pkg.JavaMetadataType, pkg.JavaMetadata{
 			ArchiveDigests: digests,
+			VirtualPath:    p.PackageSourceInfo,
```

Comment:
Looking at the test added here, I am concerned that this is changing the semantics of the `VirtualPath` -- in other words: setting values here that do not directly represent a path. This won't _necessarily_ cause a problem with things unless, for example, SPDX output needs to find a file to generate a SHA-1 hash from. Is there a way that we could search for file path patterns in the `p.PackageSourceInfo` and only include a path if we find exactly 1?

Reply:
That uncovers a potential bug; according to section 7.12.1 in the spdx spec (https://spdx.github.io/spdx-spec/v2.2.2/package-information/), the source information field has a cardinality of 0..1. The logic in syft/formats/common/spdxhelpers/source_info.go just concatenates all the paths it finds. 

In our use case, we use syft to create an SPDX-json SBOM. We then do some manipulation on it, so we need the 2 to be compatible. 
Can we merge this and raise another issue to make syft only add 1 path to source_info and then add the check when creating `VirtualPath`?

Reply:
Looking at a Syft-generated SPDX SBOM, it's not including the java metadata `VirtualPath` at all, only the location RealPath: https://github.com/anchore/syft/blob/main/syft/formats/common/spdxhelpers/source_info.go#L52

There are 2 potential paths here:
1. importing a Syft-generated SPDX document
2. importing a non-Syft-generated SPDX document

For case 1, we'd be setting `VirtualPath` to a totally different value, which is not something we would want to do. For case 2, we'd possibly like to gather as much information as we can, but this is probably less important to be the Java metadata VirtualPath but would be better suited as locations.

In both of these cases we could match all paths and add them as locations, which would reverse the current Syft encoding logic but would leave VirtualPath empty.

I don't think we're going to merge a change that is setting this value to something arbitrary, but let me ask more specifically: what is the problem you're seeing by this value not being imported?